src/orquestra/sdk/_base/_api/_task_run.py

```diff
@@ -257,3 +260,128 @@ def get_parents(self) -> t.Set["TaskRun"]:
         ]
 
         return set(parents)
+
+
+def _get_argo_backend_ids() -> (
+    t.Tuple[WorkflowRunId, t.Optional[TaskInvocationId], t.Optional[TaskRunId]]
+):
+    """
+    Get the workflow run, task invocation, and task run IDs from Argo.
+
+    Raises:
+        WorkflowRunIDNotFoundError: When the workflow run ID can't be recovered.
+    """
+
+    assert (
+        "ARGO_NODE_ID" in os.environ
+    ), "The ARGO_NODE_ID environment variable is absent."
+
+    node_id = os.environ["ARGO_NODE_ID"]
+    # Argo Workflow ID is the left part of the step ID
+    # [wf-id]-[retry-number]-[step-number]
+    wf_run_id = "-".join(node_id.split("-")[:-2])
+    task_run_id = node_id
+
+    argo_template = json.loads(os.environ["ARGO_TEMPLATE"])
+    # Looks like the template name on Argo matches our task invocation ID. Not sure how
+    # good this assumption is.
+    task_inv_id = argo_template["name"]
+
+    if wf_run_id is None:
+        raise WorkflowRunIDNotFoundError("Could not recover Workflow Run ID")
```

Comment:
**issue**: is this check ever going to pass? Even when `node_id.split("-")[:-2]` is `[]`, `"-".join([])` returns `""`

Reply:
Good point,  changed to check for empty strings instead![a3ca472](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/150/commits/a3ca472249878090ab2543adee8866aa3400aa49)